src/Server.php

```diff
@@ -159,6 +172,12 @@ public function handleRequest(ConnectionInterface $conn, Request $request)
         });
 
         $this->emit('request', array($request, $response));
+
+        if ($stream instanceof LengthLimitedStream && $contentLength === 0) {
+            // stream must emit an 'end' here, because empty data won't be emitted
+            $stream->emit('end');
+            $conn->removeListener('data', array($stream, 'handleData'));
+        }
```

Comment:
This block appears a bit unclear to me, perhaps you can clarify? The `$contentLength` variable is defined in another scope and *may* not exist here (though this *may* not cause here due to short-circuit evaluation).

Why is this listener removed (only) here?